ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -564,6 +574,15 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
       return researchPurpose.researchOutcomeList && researchPurpose.researchOutcomeList.length !== 0 ;
     }
 
+    updatePrimaryPurpose(cateogry, value) {
+      this.updateResearchPurpose(cateogry, value);
+      if (!value && !this.researchPurposeCategoriesSelected(this.state)) {
```

Comment:
a little confused by this line. `researchPurposeCategoriesSelected` takes in a research purpose right? How are we able to pass in the state object here?

Reply:
you are right not  sure what happened here but i meant to send this .state.workspace.researchPurpose